upath/core.py

```diff
@@ -123,6 +123,24 @@ class UPath(pathlib.Path, PureUPath, metaclass=UPathMeta):
     _default_accessor = _FSSpecAccessor
 
     def __new__(cls, *args, **kwargs):
+        if len(args) == 1 and isinstance(args[0], cls):
```

Comment:
Shouldn't this also check for empty `kwargs`?
```suggestion
        if len(args) == 1 and isinstance(args[0], cls) and len(kwargs) == 0:
```

Reply:
What if someone wants to do something like:
```
a = UPath("s3://bucket/folder", key="...", secret="...")
b = UPath(a, 'file')
```
I think checking `if len(args) = 1` would make this not possible. Could we instead pop the first arg if its an instance of UPath and append the rest of args to new_args? I would also say allow kwargs to be set and do something like `new_kwargs.update(kwargs)`.